File: Sorting Algo/Basics-Bubble,Insertion,Selection/selectionSort.js
/*
Move the minimum value to front
5,3,4,1,2

1,3,4,5,2
1,2,4,5,3
1,2,3,5,4
1,2,3,4,5

*/
function selectionSort(arr) {

    for (let i = 0; i < arr.length; i++) {

        let minIndex = i;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[minIndex]) {

                minIndex = j;
            }
        }
        let temp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = temp
    }
    return arr;
}

console.log(selectionSort([2, 1, 4, 3, 8, 45, 23]))

File: Sorting_1/src/SelectionSort.java

public class SelectionSort {
public static void selectionSort(int[] arr) {
    	
        int n=arr.length;
        int i=0;
        while(i<n-1){
            
            int minIdx=i;
            int j=i+1;
            while(j<n){
                if(arr[minIdx]>arr[j]){
                    minIdx=j;
                }
                j++;
            }
            int temp=arr[i];
            arr[i]=arr[minIdx];
            arr[minIdx]=temp;
            i++;
        }
        
        
    }   
	public static void main(String[] args) {
		int arr[]= {1,22,3,5,6,7};
		selectionSort(arr);
		for(int i:arr) {
			System.out.println(i);
		}
}

}
